CRM/MembershipExtras/Job/OfflineAutoRenewal/MultipleInstallmentPlan.php

```diff
@@ -91,13 +91,14 @@ public function renew() {
    *
    * The new recurring contribution will then be set to be the current recurring
    * contribution.
+   *
+   * @throws \Exception
    */
   private function createRecurringContribution() {
     $currentRecurContribution = $this->currentRecurringContribution;
     $paymentProcessorID = !empty($currentRecurContribution['payment_processor_id']) ? $currentRecurContribution['payment_processor_id'] : NULL;
 
-    $installmentReceiveDateCalculator = new InstallmentReceiveDateCalculator($currentRecurContribution);
-    $this->paymentPlanStartDate = $installmentReceiveDateCalculator->calculate($currentRecurContribution['installments'] + 1);
+    $this->paymentPlanStartDate = $this->calculateNewPeriodStartDate();
```

Comment:
the thing here is that the payment plan start date should be derived from the membership start date as you already doing, but it could differ based on cycle day. but you here setting them both to be the same